Atomic/storage.py

```diff
@@ -37,6 +61,26 @@ def reset(self):
         except:
             selinux.restorecon(root)
 
+    def add(self):
+        dss_conf = "/etc/sysconfig/docker-storage-setup"
+        dss_conf_bak = "/etc/sysconfig/docker-storage-setup" + ".bkp"
+
+        if os.path.exists(dss_conf):
+            shutil.copyfile(dss_conf, dss_conf_bak)
+        elif os.path.exists(dss_conf_bak):
+            os.remove(dss_conf_bak)
+        modify_sh_var_in_file(dss_conf, "DEVS", lambda old: sh_set_add(old, self.args.device))
+        if util.call(["docker-storage-setup"]) != 0:
+            if os.path.exists(dss_conf_bak):
+                os.rename(dss_conf_bak, dss_conf)
+            else:
+                os.remove(dss_conf)
+            util.call(["docker-storage-setup"])
+            raise ValueError("Not all devices could be added")
+        else:
+            if os.path.exists(dss_conf_bak):
+                os.remove(dss_conf_bak)
+
```

Comment:
I am thinking you wrap the entire block with a 
try:
Finally:

Then raise exceptions, when you find failures

You have this code 
if os.path.exists(dss_conf_bak):
+            os.remove(dss_conf_bak)
three times.